airbyte-metrics/reporter/src/main/java/io/airbyte/metrics/reporter/ReporterApp.java

```diff
@@ -40,11 +43,17 @@ public static void main(final String[] args) throws IOException {
 
     try (final DSLContext dslContext = DSLContextFactory.create(dataSource, SQLDialect.POSTGRES)) {
 
+      final Flyway flyway = FlywayFactory.create(dataSource, ReporterApp.class.getSimpleName(),
+          ConfigsDatabaseMigrator.DB_IDENTIFIER, ConfigsDatabaseMigrator.MIGRATION_FILE_LOCATION);
+
       // Ensure that the database resources are closed on application shutdown
       CloseableShutdownHook.registerRuntimeShutdownHook(dataSource, dslContext);
 
-      configDatabase = new ConfigsDatabaseInstance(dslContext)
-          .getInitialized();
+      // Ensure that the Configuration database is available
+      DatabaseCheckFactory.createConfigsDatabaseMigrationCheck(dslContext, flyway, configs.getConfigsDatabaseMinimumFlywayMigrationVersion(),
+          configs.getConfigsDatabaseInitializationTimeoutMs()).check();
+
+      configDatabase = new Database(dslContext);
```

Comment:
Related to my previous comment - this is the order I expected to see: we perform the database check first, and then create the `Database` object after that

Reply:
@lmossman Agreed.  I can do another pass at the Bootloader to see if I can make changes to avoid that, but I am hesitant to mess with too much code, if I can help it, especially given that we will be throwing away these *App classes.